documentation/stp/test-plan.md

```diff
@@ -0,0 +1,62 @@
+# PACE Test Plan
+
+## Overview
+
+Testing of Horace covers areas:
+
+- return correct data for specific (small) inputs
+- handle edge cases correctly
+- handle error cases correctly
+- behave identically running on one or more thread (shared memory)
+- behave identically running on one or mode nodes
+- high-level functions behave identically when executed with realistic data sizes
+- performance as a function of both file size and hardware
+
+## Development and System Test Plan
+
+Tests will be written for every story as it is implemented.
+
+Full unit and software component integration testing will be automatically executed for regression purposes when code is merged and committed into the main development branch in the code repository. This will be complemented by manual testing as necessary.
+
+The completion of all testing activities is a part of the Definition of Done. 
+
+For each story and for every Sprint the following test strategy will be followed:
+
+-	Automated unit tests will be written by the developers for all code that is developed except where impractical.
+-	Automated integration tests will be written (using the unit test framework) by the developers where practical for collections of integrated software components (e.g. Matlab calling compiled C++ implementations).
+
+In addition:
+
+-	Automated system tests will be written by the developers to script the execution of a number of standard end-to-end workflows to measure performance on representative data sets and hardware.
+
+### Unit Testing
+
+All (new) code will be tested at UNIT level
+
+- Where code is parallelized tests must be extended to cover execution on 1 and N (>1) threads to verify behaviour is unchanged
+- C++ code will be unit tested with CXX
```

Comment:
CXXTest has not had a release in 5 or 6 years https://sourceforge.net/p/cxxtest/blog/. Some more modern frameworks are [googletest](https://github.com/google/googletest) or, my personal preference (which I've used before), [Catch2](https://github.com/catchorg/Catch2). Catch has the advantage of being header only so is pretty painless to set up

Reply:
The decision of which specific C++ test framework is covered under  #32. I can replace "CXX" in this document with whichever decision is reached there.

While Catch2 may be easier to setup, googletest is being used by the IBEX team and the googletest mocking framework by Mantid.